### PR TITLE
west: use image-1's size as slot size in imgtool

### PR DESCRIPTION
**Pull request by utzig**
When MCUBoot is built using the swap-move strategy, the secondary slot can be one sector smaller than the primary slot, because the primary slot's last "useful" image sector must be "reserved" for the move operation; this impacts the generation of padded images, so when signing an image, the proper size to use is the secondary slot's.

### CODE REVIEW

File: scripts/west_commands/sign.py [hunk 1]
```diff
@@ -350,12 +359,17 @@ def edt_flash_params(flash):
         if align == 0:
             log.die('expected nonzero flash alignment, but got '
                     'DT flash device write-block-size {}'.format(align))
-        reg = image_0.regs[0]
+        reg = images['image-0'].regs[0]
+        if reg.addr == 0:
+            log.die('expected nonzero slot address, but got '
+                    'DT image-0 partition address {}'.format(reg.addr))
+        addr = reg.addr
```
Comment (mbolivar-nordic) [inline scripts/west_commands/sign.py:366]: This change seems unrelated to using image-1's size. Why is it being added?

**Review by mbolivar-nordic — COMMENTED** [review]
Seems reasonable, thanks for catching this edge case.

I think I heard of some discussion of using mcuboot without an image-1 partition in some configurations. Did I imagine that? If not, should this be falling back on image-0's size if image-1 does not exist?

File: scripts/west_commands/sign.py (hunk 1 above)
Comment (utzig) [reply scripts/west_commands/sign.py]: This was just added because `addr` and `size` in that commit changed to arrive from different places, and it seemed like a good idea to test it then, but in retrospect since `addr` is not used by imgtool, only by printing routines, there is not point in testing it, so I removed this in the fixed commit.

**Comment by utzig** [comment]
> I think I heard of some discussion of using mcuboot without an image-1 partition in some configurations. Did I imagine that? If not, should this be falling back on image-0's size if image-1 does not exist?

Yeah, I totally forgot about it. The updated commit tries to read `size` from `image-1` if that is available, otherwise falls back to `image-0`.

File: scripts/west_commands/sign.py [hunk 2]
```diff
@@ -350,12 +350,22 @@ def edt_flash_params(flash):
...
-            log.die('expected nonzero slot size, but got '
-                    'DT image-0 partition size {}'.format(reg.size))
 
-        return (align, reg.addr, reg.size)
+        # The partitions node, and its subnode, must provide
+        # the size of image-1 or image-0 partition via the regs property.
+        image_key = 'image-1' if 'image-1' in images else 'image-0'
+        if not images[image_key].regs:
+            log.die('DT zephyr,flash node has no regs property;',
+                    "can't determine size of {}".format(image_key))
```
Comment (mbolivar-nordic) [inline scripts/west_commands/sign.py:359]: Since this is an error message that users will see, I think this should be changed to refer to the exact node which is missing `regs`. Right now, it's referring to a different node, which might be confusing for people.

The chosen `zephyr,flash` node is something like `/soc/flash-controller@foo/flash@0`. That node is not missing a `regs` property.

Its grandchild node referring to a partition (something like `/soc/flash-controller@foo/flash@0/partitions/partition@bar`) is actually what is missing a `regs`.

```suggestion
            log.die(f'{image_key} flash partition has no regs property;',
                    "can't determine size of image")
```

**Review by mbolivar-nordic — COMMENTED** [review]
Thanks for adding the fallback to image-0. I just have one concern about an error message, then I think this looks good to go.

File: scripts/west_commands/sign.py (hunk 2 above)
Comment (utzig) [reply scripts/west_commands/sign.py]: Updated as suggested.

**Review by mbolivar-nordic — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Review by nvlsianpu — APPROVED** [review]
